src/main/java/org/peergos/HostBuilder.java

```diff
@@ -82,12 +83,33 @@ public HostBuilder generateIdentity() {
         return setPrivKey(Ed25519Kt.generateEd25519KeyPair().getFirst());
     }
 
+    public HostBuilder setIdentity(String base64PrivKey) {
+        byte[] privKey = Base64.decodeBase64(base64PrivKey);
+        return setPrivKey(Ed25519Kt.unmarshalEd25519PrivateKey(privKey));
+    }
+
+
+
     public HostBuilder setPrivKey(PrivKey privKey) {
         this.privKey = privKey;
         this.peerId = PeerId.fromPubKey(privKey.publicKey());
         return this;
     }
 
+    public static HostBuilder build(String privKeyBase64, int listenPort, ProviderStore providers, RecordStore records, Blockstore blocks) {
```

Comment:
Why do we need this method with hard coded protocols?

Reply:
copied relevant code to Server.java

Reply:
So this method can go?

Reply:
yes, done